src/components/AddRestaurantModal.jsx

```diff
@@ -1,43 +1,56 @@
 import '../css/Modal.css';
+import { useState } from 'react';
 
-function AddRestaurantModal() {
+function AddRestaurantModal({ onChangeModal, currentRestaurants, handleRestaurant }) {
+    const [category, setCategory] = useState('');
+    const [name, setName] = useState('');
+    const [description, setDescription] = useState('');
+
+    const handleFomSubmit = (e) => {
+        e.preventDefault();
+        const newRestaurant = {
+            id: Date.now(),
+            category,
+            name,
+            description,
+        };
+        handleRestaurant([...currentRestaurants, newRestaurant]);
```

Comment:
```tsx
const [restaurants, setRestaurants] = useState(restaurantsData);

<AddRestaurantModal currentRestaurants={restaurants} handleRestaurant={setRestaurants} />
```
지금은 상위에서 선언된 현재 음식점 리스트(`currentRestaurants`)와 음식점 리스트를 추가할 수 있는 setState 함수(`handleRestaurant`)를 prop로 받아서 새로운 음식점을 추가하고 있는데, 함수형 업데이트를 사용한다면 `handleRestaurant`함수만으로 이전과 같은 동작을 하고 불변성을 지킬 수 있을 것 같아요!🙃

```suggestion
        // currentRestaurants는 함수형 업데이트로 이전 상태 값을 참조하기 때문에 prop에서 제거해도 됨
        handleRestaurant((currentRestaurants) => [...currentRestaurants, newRestaurant]);
```

[이전 state를 기반으로 state 업데이트하기 - React 공식 문서](https://ko.react.dev/reference/react/useState#updating-state-based-on-the-previous-state)